### PR TITLE
JENKINS-66735: Whitelist some common signatures and additional signatures from `CauseOfInterruption`

### PR DESCRIPTION
**Pull request by haridsv**
The change whitelists a few signatures listed in [JENKINS-66735](https://issues.jenkins.io/browse/JENKINS-66735)

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Review by jglick — APPROVED** [review]
All look harmless.

**Comment by haridsv** [comment]
@jglick Thank you for the approval. I just added one more HashMap constructor, I hope that doesn't change your stance on approval.

File: src/main/resources/org/jenkinsci/plugins/scriptsecurity/sandbox/whitelists/jenkins-whitelist [hunk 1]
```diff
@@ -34,3 +34,5 @@ field hudson.scm.EditType EDIT
 method hudson.scm.EditType getName
 method hudson.tools.ToolInstallation getHome
 method hudson.tools.ToolInstallation getName
+method jenkins.model.CauseOfInterruption getShortDescription
+method jenkins.model.CauseOfInterruption.UserInterruption getUserId
```
Comment (jglick) [inline src/main/resources/org/jenkinsci/plugins/scriptsecurity/sandbox/whitelists/jenkins-whitelist:38]: Did you actually test this? I think you meant
```suggestion
method jenkins.model.CauseOfInterruption$UserInterruption getUserId
```

Comment (haridsv) [reply src/main/resources/org/jenkinsci/plugins/scriptsecurity/sandbox/whitelists/jenkins-whitelist:38]: Thanks a lot for catching the typo. I didn't start making the changes that need these whitelists yet, but I was expecting the test to fail for any typos so with the test passing, I didn't think that it needed to be tested. Could there be a flaw in the test?

Comment (jglick) [reply src/main/resources/org/jenkinsci/plugins/scriptsecurity/sandbox/whitelists/jenkins-whitelist]: https://github.com/jenkinsci/script-security-plugin/blob/b4f85af4477cfa0ee27a03ed0baf439788fb4c6f/src/test/java/org/jenkinsci/plugins/scriptsecurity/sandbox/whitelists/StaticWhitelistTest.java#L105 is supposed to be catching this. I will need to take a look.

Comment (jglick) [reply src/main/resources/org/jenkinsci/plugins/scriptsecurity/sandbox/whitelists/jenkins-whitelist]: Ah right. https://github.com/jenkinsci/script-security-plugin/blob/c3d9d689368dcb6b7ae5cfdba4c7537fc8680194/src/main/java/org/jenkinsci/plugins/scriptsecurity/sandbox/whitelists/EnumeratingWhitelist.java#L221

**Pull request merged**